floss/language/go/extract.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+
+import re
+import sys
+import struct
+import logging
+import argparse
+from typing import List, Iterable, Optional
+
+import pefile
+
+from floss.results import StaticString, StringEncoding
+
+logger = logging.getLogger(__name__)
+
+MIN_STR_LEN = 6
+
+
+def extract_go_strings(
+    sample: str,
+    min_length,
+) -> Iterable[StaticString]:
+    """
+    Get Go strings from a PE file.
+    Reference: https://github.com/mandiant/flare-floss/issues/779
+    """
+
+    try:
+        pe = pefile.PE(sample)
+    except pefile.PEFormatError as err:
+        logger.debug(f"invalid PE file: {err}")
+        raise ValueError("Invalid PE header")
+
+    if pe.FILE_HEADER.Machine == pefile.MACHINE_TYPE["IMAGE_FILE_MACHINE_AMD64"]:
+        """
+        .data:0000000000770F20 3D 68 60 00 00 00+off_770F20      dq offset aString
+        .data:0000000000770F28 15                                db  15h
+        .data:0000000000770F29 00                                db    0
+        """
+        alignment = 0x10  # 16
+        fmt = "<QQ"
+
+        # See https://github.com/mandiant/flare-floss/issues/805#issuecomment-1590472813 for regex explanation
```

Comment:
the comment doesn't really explain the regex and I really think we should have it inline here showing hex bytes and instructions targeted by the patterns, for example:

```python
"""
48 BA 41 64 64 44 6C 6C 44 69 mov     rdx, 'iDllDddA'
48 89 94 24 AD 00 00 00       mov     [rsp+158h+var_AB], rdx
48 BA 72 65 63 74 6F 72 79 00 mov     rdx, 'yrotcer'
48 89 94 24 B5 00 00 00       mov     [rsp+158h+var_A3], rdx
"""

```